lib/OpenQA/Schema/Result/JobModules.pm

```diff
@@ -90,10 +90,9 @@ sub results {
     return {} unless my $dir = $self->job->result_dir;
     my $name = $self->name;
 
-    return {} unless -r (my $file = path($dir, "details-$name.json"));
-
-    my $json_data = $file->slurp;
-    die qq{Malformed JSON file "$file": $@} unless my $json = eval { decode_json($json_data) };
+    my $file = path($dir, "details-$name.json");
+    return {} unless my $json_data = eval { $file->slurp };
```

Comment:
This would consider an empty file an error. In addition, the previous code only returned {} in case the file wasn't readable and now we treat all errors like this.

Reply:
> Why would this treat an empty file as an error?

I mean it would return an empty hash (which is the error handling here).

I guess the change is generally ok but it slightly changes what errors are handled how (instead of just fixing the race condition).